src/dftbp/timedep/timeprop.F90

```diff
@@ -2883,36 +2886,32 @@ subroutine tdPopulInit(this, Eiginv, EiginvAdj, eigvecsReal, eigvecsCplx)
     complex(dp), allocatable :: T2(:,:), T3(:,:)
     integer :: iOrb
 
-    allocate(T2(this%nOrbs, this%nOrbs), T3(this%nOrbs, this%nOrbs))
-
+    allocate(T2(this%nOrbs, this%nOrbs))
+    allocate(T3(this%nOrbs, this%nOrbs))
     if (this%tRealHS) then
-      T2 = cmplx(eigvecsReal, 0, dp)
+      T2(:,:) = cmplx(eigvecsReal, kind=dp)
```

Comment:
Is this a compiler work around or just cleaning up source code?

Reply:
It prohibits reallocation, as `T2` is allocatable. Not sure, whether the reallocation was part of the problem, but probably it was, so I'd keep it.